xarray/core/dataarray.py

```diff
@@ -871,13 +871,19 @@ def chunks(self) -> Optional[Tuple[Tuple[int, ...], ...]]:
         """
         return self.variable.chunks
 
-    def chunk(self, chunks: Union[
-        None, int, Tuple[int, ...], Tuple[Tuple[int, ...], ...],
-        Mapping[Hashable, Union[None, int, Tuple[int, ...]]],
-    ] = None,
-            name_prefix: str = 'xarray-',
-            token: Optional[str] = None,
-            lock: bool = False) -> 'DataArray':
+    def chunk(
+        self,
+        chunks: Union[
+            None,
+            Number,
```

Comment:
Last I checked, type checkers were much happier with `int` rather than `numbers.Number`. Maybe mypy has been updated since then, though....

Reply:
dask will happily accept float, numpy.int64, etc. and transparently cast them to int.
Should we decide that dask is excessively nice and the _intended_ input must strictly be int?

Reply:
I vote to lean towards wider input bounds - it much more annoying for users to have false negatives than false positives; so in this case `Number` assuming mypy understands that

Reply:
OK, let's give this a try. We can always change later, that's not a big deal.

One other minor note -- it would be great to define a few type aliases for these longer types. The line wrapping you've done here helps, but this is still very long to write in a function signature.